pubsub/server_test.go

```diff
@@ -190,12 +196,32 @@ func TestServerRead(t *testing.T) {
 	<-serverDone
 }
 
+// waitTimeout waits for the waitgroup for the specified max timeout.
+// Returns true if waiting timed out.
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		wg.Wait()
+	}()
+	select {
+	case <-c:
+		return false // completed normally
+	case <-time.After(timeout):
+		return true // timed out
+	}
+}
+
+func optimisticEventuallity(require *require.Assertions, condition func() bool, waitFor time.Duration, tick time.Duration, msgAndArgs ...interface{}) { //nolint:unparam
+	if !condition() {
+		require.Eventually(condition, waitFor, tick, msgAndArgs...)
+	}
+}
```

Comment:
is the purpose of this function just to make sure that we run the condition first before waiting?

Reply:
yes, exactly. I'll add a comment above.